lib/engine-native/src/artifact.rs

```diff
@@ -545,13 +546,19 @@ impl NativeArtifact {
         })?;
         let metadata_slice: &'static [u8] =
             slice::from_raw_parts(&size[10] as *const u8, metadata_len as usize);
-        let metadata: ModuleMetadata = bincode::deserialize(metadata_slice)
-            .map_err(|e| DeserializeError::CorruptedBinary(format!("{:?}", e)))?;
+
+        let metadata = ModuleMetadata::deserialize(metadata_slice)?;
+
         let mut engine_inner = engine.inner_mut();
 
         Self::from_parts(&mut engine_inner, metadata, shared_path, lib)
             .map_err(DeserializeError::Compiler)
     }
+
+    /// Used in test deserialize metadata is correct
+    pub fn metadata(&self) -> &ModuleMetadata {
+        &self.metadata
+    }
```

Comment:
```suggestion
    /// Used in test deserialize metadata is correct
    #[cfg(test)]
    pub fn metadata(&self) -> &ModuleMetadata {
        &self.metadata
    }
```

If this function is only used for testing purposes, maybe it's better that way, thoughts?

Reply:
Tried, however rust seems not happy to compile, even though it's only used in test:
```
error[E0599]: no method named `metadata` found for struct `NativeArtifact` in the current scope
   --> lib/api/src/store.rs:153:29
    |
153 |         assert_eq!(artifact.metadata(), artifact_deserialized.metadata());
    |                             ^^^^^^^^ private field, not a method

error[E0599]: no method named `metadata` found for struct `NativeArtifact` in the current scope
   --> lib/api/src/store.rs:153:63
    |
153 |         assert_eq!(artifact.metadata(), artifact_deserialized.metadata());
    |                                                               ^^^^^^^^ private field, not a method

error: aborting due to 2 previous errors
```
(lib/api/src/store.rs:153:29 is in `#[cfg(test)]` and `#[test]` 🤔 

I'll probably drop this, given will test the artifact's enclosing module == deserialized module, by deriving Eq in artifact, so doesn't need pub access artifact.metadata anymore